cmake/api_macros.cmake

```diff
@@ -191,6 +191,21 @@ macro(OPENDDS_TARGET_SOURCES target)
     list(APPEND _tao_options "-SS")
   endif()
 
+  list(LENGTH ${CMAKE_CXX_COMPILER} cxx_compiler_length)
+
+  if(${cxx_compiler_length} LESS 1)
```

Comment:
```suggestion
  if(${cxx_compiler_length} EQUAL 1)
```

Reply:
It's 0 or undefined if it isn't a list.  

If CMAKE_CCX_COMPILER is "path/to/cl/exe", then it is 0 or undefined.  

If CMAKE_CXX_COMPILER is "path/to;cl/exe" then it is length 2.

If it is a list of any length (meaning has a ';') we don't want it in the options because it will get put in as something weird like "-Yp,path/to;cl/exe"  which will cause problems.

Reply:
I totally agree with not passing it if it's a list, that's what I'm trying to avoid. This is related to comment about `list(LENGTH...)`, even though I didn't see that at first. If that's fixed then going to be 1 if `CMAKE_CXX_COMPILER` is normal.